### PR TITLE
Remove add_legacy_key and related methods

### PR DESCRIPTION
**Pull request by Fryguy**
recrypt is the only method that uses the legacy key, so making a
parameter to that method greatly simplifies the interface.

### CODE REVIEW

**Review by kbrock — COMMENTED** [review]
Looking good Lewis

I would have preferred a non lockstep approach... but I'm sure that was noted from my previous merge.

could you rebase, link to the core PR and run a cross repo? definitely automate, more extensions, and core - not sure about others

File: lib/manageiq/password.rb [hunk 1]
```diff
@@ -102,52 +90,29 @@ def self.try_encrypt(str)
...
     end
 
-    def self.v2_key
-      keys["v2"]
-    end
-
-    def self.load_v2_key
-      load_key_file("v2_key") || begin
+    def self.key
+      @@key ||= load_key_file("v2_key") || begin
```
[the remark this answers is not in the document]
Comment (Fryguy) [reply lib/manageiq/password.rb:115]: probably not, but we've always had it as a warning...not sure where to put it though.

**Comment by Fryguy** [comment]
Rebased now that #14 is merged.

**Comment by kbrock** [comment]
@Fryguy this looks nice. do we first need to remove `add_legacy_key` and recrypt from miq core? how do we coordinate this change?

**Comment by Fryguy** [comment]
My plan is to just do a lot of breaking changes and then cut a 1.0 release.  Then after it's cut I change manageiq core to accommodate.

**Comment by Fryguy** [comment]
@kbrock This is rebased and ready to go.

**Pull request merged**